### ISSUE TITLE
Make Derivative axis-free

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaApproximation/ContinuumArrays.jl#22]
I think `Derivative` should behave like `UniformScaling` and not require axes.

**Comment by jagot** [comment]
:+1:, but how does that scale to multiple dimensions? I.e. how do you distinguish between partial derivatives, along e.g.`x` only, and the gradient operator?

**Comment by dlfivefifty** [comment]
Good question. I suppose we could have:
```julia
D = Derivative()
D2 = D^2 # stored lazily as `applied(^,D,2)`
Dx = D ⊗ I # stored lazily as `applied(⊗,D, I)`
Dy = I ⊗ D  # stored lazily as `applied(⊗,I,D)`
∇ = [Dx; Dy] # stored lazily as `applied(vcat,Dx,Dy)`
```

Alternatively, we could try to do things "properly" with differential forms...

**Comment by jagot** [comment]
I guess the ⊗ approach works for curvilinear coordinates (polar/spherical coordinates, etc), but if we ever want to do differential geometry, i.e. with a spatially varying metric, we need differential forms (?)

**Comment by dlfivefifty** [comment]
It's interesting to think what differentiable forms would look like in quasi-array language: I'm not aware of finite-dimensional analogues of differential forms. Though we'll still need gradients and partial derivatives anyways so maybe best not to overthink it: just have `Gradient` and `Laplacian` types?

**Comment by jagot** [comment]
Yes, I agree that `Gradient` and `Laplacian` will be enough for multiple dimensions, for now. I guess that `PartialDerivative` will still need to be attached to an axis though?

**Comment by dlfivefifty** [comment]
Yes, since partial derivatives are inherently coordinate dependent they probably don’t need their own type, and the construction above is fine
